src/components/AssetIcon/AssetIcon.tsx

```diff
@@ -73,18 +76,65 @@ export const MultipleAssetLogo = ({
 export const AssetLogo = ({ id }: { id?: string }) => {
   const { t } = useTranslation()
   const { getAsset, getErc20, isErc20 } = useAssets()
-
-  const { getIsWhiteListed } = useExternalAssetsWhiteList()
+  const { getExternalAssetMetadata, isInitialized } = useExternalAssetsMetadata(
+    useShallow((state) => ({
+      getExternalAssetMetadata: state.getExternalAssetMetadata,
+      isInitialized: state.isInitialized,
+    })),
+  )
```

Comment:
Nitpick,  but in future you can use `pick` to shorten this

```suggestion
const { getExternalAssetMetadata, isInitialized } = useExternalAssetsMetadata(
  useShallow((state) => pick(state, ["getExternalAssetMetadata", "isInitialized"])),
)
```

with Remeda, it could be even shorter
```suggestion
const { getExternalAssetMetadata, isInitialized } = useExternalAssetsMetadata(
  useShallow(pick(state, ["getExternalAssetMetadata", "isInitialized"])),
)
```

Reply:
we don't have remeda in this version :)